content/blog/getting-started-with-java-development-2023.asciidoc

```diff
@@ -0,0 +1,211 @@
+This is a tool which allows you to install software development kits (SDKs) such as Java's JDK.
+You can also update your installed SDK versions and easily switch between multiple versions.
+
+If you have SDKMan installed, obtaining the current Eclipse Temurin build of Java 17 is as simple as running the following in your shell:
+
+[source,bash,linenums=true]
+----
+$ sdk install java 17.0.5-tem # Install
+$ sdk use java 17.0.5-tem # Activate
+$ java --version # Verify version
+
+openjdk 17.0.5 2022-10-18
+...
+----
+
+== 💡 Your First Java Program
+
+Having installed Java, it's time to write your first Java program.
+Java is an object-oriented language (with some functional flavour),
+hence everything in a Java program is defined in form of classes, which have fields (representing their state) and methods (the behavior operating on that state).
+The canonical "Hello World" example in Java looks like this:
+
+[source,java,linenums=true]
+----
+public class HelloWorld { // <1>
+  public static void main(String... args) { // <2>
+    System.out.println("Hello world!"); // <3>
+  }
+}
+----
+<1> The class `HelloWorld` must be specified in a source file named _HelloWorld.java_
+<2> The `main()` method is the entry point into a Java program
+<3> The `println()` method prints the given text to standard out
+
+Java source code is compiled into class files which then are loaded into the JVM and executed.
+Normally, this is done in two steps: first running the compiler _javac_, then executing the program using the _java_ binary.
+For quick testing and exploring, both steps can be combined, so you can execute your "Hello World" program like this:
+
+[source,bash,linenums=true]
+----
+$ java HelloWorld.java
+Hello world!
+----
+
+For exploring Java in a quick and iterative mode,
+it provides *jshell*, an interactive Read-Evaluate-Print Loop (REPL).
+You can use it for running expressions and statements without defining a surrounding method or class,
+simplifying "Hello World" quite a bit:
+
+[source,bash,linenums=true]
+----
+$ jshell
+
+jshell> System.out.println("Hello World");
+Hello World
+----
+
+Similar to jshell, but quite a bit more fancier is *https://www.jbang.dev/[jbang]*,
+which for instance allows you to easily pull in 3rd party libraries into your single source file Java programs.
+
+== 👷‍♀️ Build Tool
+
+Once you go beyond the basics of manually compiling and running a set of Java classes,
+you'll need a build tool.
+It will help you not only with compiling your code,
+but also with managing dependencies (i.e. 3rd party libraries you are using),
+testing your application, assembling the output artifacts (e.g. a JAR file with your program), and much more.
+There are plug-ins for finding common bugs, auto-formatting your code, etc.
+Commonly used build tool options for Java include Apache Maven, Gradle, and Bazel.
+
+My recommendation is to stick with *Maven* for the beginning;
```

Comment:
Could be worth mentioning that the build tools share same (maven) registry? So regardless what build tool one choses, available 3rd party libs are the same?

Reply:
I first assumed you'd refer to the local (Maven) repo, but I believe this is about Maven Central actually? If so, yes, that's a very good idea, and I have added that already.